### ISSUE TITLE
SelfService users bypassing background agent

### ISSUE THREAD

**Issue by lukemgriffith** [chocolatey/chocolatey-licensed-issues#34]
### What You Are Seeing?
Intermittent users being unable to use the background service, Chocolatey log shows useBackgroundAgent enabled, but does not respect this. 

I'd be keen to create a pester test to identify if a user context is able to access the background agent, from looking at the code I can see that SessionName needs to be set specifically, can you advise what else needs to be aligned for the choco agent to attempt to pass the request back to the self  service agent? 

### What is Expected?
User to be passed to the background service

### How Did You Get This To Happen? (Steps to Reproduce)
Currently struggling to determine how to reproduce this. We've seen times where the user does not have an $ENV:SESSIONNAME, and times when the user does and its "Console" / "*RDP*"

Currently trying to collate a number of logs for further debugging, and so we can provide them to your selves. 

### Output Log

Extract from what we have so far... i'll get more 

2018-01-10 17:23:08,433 19236 [INFO ] - ============================================================
2018-01-10 17:23:08,549 19236 [DEBUG] - [Background] Configuring background service.
2018-01-10 17:23:08,553 19236 [DEBUG] - [Background] Not using background service, user is remote session (not remote
desktop).

**Comment by ferventcoder** [comment]
Howdy @lukemgriffith, I think this stems from that other fix we put in about seeing calls coming in from winrm and not letting them go through the background service. We are going to change that over to a feature you opt into explicitly (and note that it is experimental - as it has issues like this). If you reach out to the support email, we will get you a build to test with.

**Comment by lukemgriffith** [comment]
Just to follow up on this, its not effecting winrm users but interactive users who are logged in via console, or RDP.

They're not administrators, and we've enabled UseBackgroundServiceWithNonAdministratorsOnly.

This one case, the logs are showing that the user is remote, when the information is showing the user is interactive. The user has a direct console connection to the machine in question - unsure why its showing 'remote'.

This this might be due to the reliance on the environment session name variable. I've witnessed occasions where this variable is not populated. I've enabled environment logging and am waiting for the user to re-run the install so I can capture more information. 


```
  C:\ProgramData\chocolatey\logs\chocolatey.1.log:12636:Information.Is64BitOperatingSystem='True'|
  C:\ProgramData\chocolatey\logs\chocolatey.1.log:12637:Information.Is64BitProcess='True'|Information.IsInteractive='True'|
  C:\ProgramData\chocolatey\logs\chocolatey.1.log:12638:Information.UserName='[redact]'|Information.UserDomainName='[redact]'|
  C:\ProgramData\chocolatey\logs\chocolatey.1.log:12639:Information.IsUserAdministrator='False'|
  C:\ProgramData\chocolatey\logs\chocolatey.1.log:12640:Information.IsUserSystemAccount='False'|
  C:\ProgramData\chocolatey\logs\chocolatey.1.log:12641:Information.IsUserRemoteDesktop='False'|
  C:\ProgramData\chocolatey\logs\chocolatey.1.log:12642:Information.IsUserRemote='True'|
  C:\ProgramData\chocolatey\logs\chocolatey.1.log:12643:Information.IsProcessElevated='False'|
  C:\ProgramData\chocolatey\logs\chocolatey.1.log:12644:Information.IsLicensedVersion='True'|
  C:\ProgramData\chocolatey\logs\chocolatey.1.log:12645:Information.LicenseType='Business'|
```

**Comment by ferventcoder** [comment]
> This this might be due to the reliance on the environment session name variable.

It is exactly that. Unfortunately we have not found a direct way to determine someone is using WinRM. We thought the empty session name would give us remoting sessions but it appears that is not the case as it has too many false positives (especially in Windows 10). So we are backing that out to an experimental feature.

**Comment by ferventcoder** [comment]
@lukemgriffith we've sent over a fixed version for you to take a look at through ZenDesk. Give that a shot and see if it corrects the issue - we also have another customer verifying this as well currently. Once we have verification, we are going to issue a release.

**Comment by ferventcoder** [comment]
This has been fixed for 1.12.11 and will be released soon.
